pkg/operator/controller/status.go

```diff
@@ -95,29 +90,35 @@ func (r *reconciler) syncOperatorStatus() error {
 
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState() (*corev1.Namespace, []operatorv1.DNS, []appsv1.DaemonSet, error) {
+func (r *reconciler) getOperatorState() (*corev1.Namespace, []operatorv1.DNS, error) {
 	ns := manifests.DNSNamespace()
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, ns); err != nil {
 		if errors.IsNotFound(err) {
-			return nil, nil, nil, nil
+			return nil, nil, nil
 		}
-		return nil, nil, nil, fmt.Errorf("failed to get namespace %s: %v", ns.Name, err)
+		return nil, nil, fmt.Errorf("failed to get namespace %s: %v", ns.Name, err)
 	}
 
 	dnsList := &operatorv1.DNSList{}
 	if err := r.client.List(context.TODO(), dnsList); err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to list dnses: %v", err)
+		return nil, nil, fmt.Errorf("failed to list dnses: %v", err)
 	}
 
-	daemonsetList := &appsv1.DaemonSetList{}
-	if err := r.client.List(context.TODO(), daemonsetList, client.InNamespace(ns.Name)); err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to list daemonsets: %v", err)
-	}
-	return ns, dnsList.Items, daemonsetList.Items, nil
+	return ns, dnsList.Items, nil
+}
+
+// computeOperatorStatusConditions computes the operator's current state.
+func computeOperatorStatusConditions(conditions []configv1.ClusterOperatorStatusCondition, ns *corev1.Namespace,
+	dnses []operatorv1.DNS) []configv1.ClusterOperatorStatusCondition {
+	conditions = computeOperatorDegradedCondition(conditions, ns)
+	conditions = computeOperatorProgressingCondition(conditions, dnses)
+	conditions = computeOperatorAvailableCondition(conditions, dnses)
```

Comment:
Could we use the same `conditions = setStatusCondition(conditions, computeFooCondition(...))` pattern here that you implemented in `computeDNSStatusConditions`?

Reply:
Sure. I'll push an update to the PR in a few minutes. Thanks for all the great feedback!